File: frontend/web/services/workflows/delete.ts
import { ServiceReturn } from "@/types/api";
import { axiosInstance } from "@/services";
import { SERVICE_ERROR_UNKNOWN } from "@/config/services";

export const deleteOne = async (workflowId: string): Promise<ServiceReturn<void>> => {
	try {
		const response = await axiosInstance.delete<void>(`/workflows/${workflowId}`);
		return { data: response.data, error: undefined };
	} catch (error) {
		return { data: null, error: SERVICE_ERROR_UNKNOWN };
	}
};

export const deleteBulk = async (workflowIds: string[]): Promise<ServiceReturn<void>> => {
	try {
		const response = await axiosInstance.delete<void>(`/workflows/bulk}`, { data: { workflows: workflowIds } });
		return { data: response.data, error: undefined };
	} catch (error) {
		return { data: null, error: SERVICE_ERROR_UNKNOWN };
	}
};

File: frontend/mobile/lib/services/workflows/delete.dart
import 'package:area_mobile/services/dio.dart';
import 'package:area_mobile/types/services.dart';
import 'package:area_mobile/types/workflows/workflows.dart';

Future<ServiceReturn<int>> delete(Workflow old) async {
  try {
    final response = await dio.delete<Workflow>('/workflows/${old.id}');
    print("status code is :");
    print(response.statusCode);
    return ServiceReturn(data: response.statusCode);
  } catch (e) {
    print("error is :");
    print(e.toString());
    return ServiceReturn(error: e.toString());
  }
}
